src/main/java/org/broadinstitute/consent/http/service/DarCollectionService.java

```diff
@@ -98,4 +104,22 @@ public DarCollection updateDarCollection(Integer collectionId, User user) {
     darCollectionDAO.updateDarCollection(collectionId, user.getDacUserId(), new Date());
     return getByCollectionId(collectionId);
   }
+
+  //Writing this method here so it can be testable and reused for collection resources
+  //This method will be expanded to account for dataset binning once roles come into play
+  //For now it's functionality is simple, fetch the datasets and attach it to the collection
+  //DTO class is used to to it's dataUse attribute which will be needed for binning functionality (to be added soon)
+  public void addDatasetsToCollections(List<DarCollection> collections) {
+    for (DarCollection collection : collections) {
+      List<Integer> datasetIds = new ArrayList<Integer>();
+      for (DataAccessRequest dar : collection.getDars()) {
+        datasetIds.add(dar.getData().getDatasetIds().get(0));
+      }
+      List<Integer> distinctIds = datasetIds.stream().distinct().collect(Collectors.toList());
+      // Dataset call is being done seperatly to keep collection queries and collection mappers/reducers from getting messy
+      Set<DatasetDTO> datasets = datasetDAO.findDatasetDTOByIdList(distinctIds);
```

Comment:
Is there a strong reason to use `DatasetDTO` over `DataSet`? Ideally, we should use real models over DTOs. DTOs are generally meant not meant to represent the primary model data, but rather a manipulated form of it. In the long term, I'd really like to see the DTO go away.

Reply:
Filed https://broadworkbench.atlassian.net/browse/DUOS-1442 to cover this refactor

Reply:
It looks like the `DataUse` is the one thing we need that `DataSet` doesn't have. Can we add that as a field to `DataSet` and query for it via the join from `consentassociations` -> `consent`?